src/main/java/com/commercetools/emailprocessor/email/EmailProcessor.java

```diff
@@ -0,0 +1,91 @@
+package com.commercetools.emailprocessor.email;
+
+import com.commercetools.emailprocessor.model.Email;
+import com.commercetools.emailprocessor.model.Statistics;
+import com.commercetools.emailprocessor.model.TenantConfiguration;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.queries.CustomObjectQuery;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+
+public class EmailProcessor {
+
+
+    private static final Logger LOG = LoggerFactory.getLogger(EmailProcessor.class);
+    private static final String PARAM_EMAIL_ID = "emailid";
+    private static final String PARAM_TENANT_ID = "tenantid";
+    private static final String CONTAINER_ID = "unprocessedEmail";
+  ;
+    private static final String EMAIL_STATUS_PENDING = "pending";
+
+    public EmailProcessor() {
+    }
+
+    /**
+     * Iterate through all Email objects and triggers the webhook for each  pending email object
+     *
+     * @param tenantConfiguration Configuration of a tenant
+     * @return Statics of the sended emails
+     */
+
+    public Statistics processEmails(TenantConfiguration tenantConfiguration) {
+
+        SphereClient client = tenantConfiguration.getSphereClient();
+        CustomObjectQuery<Email> query = CustomObjectQuery.of(Email.class);
+        query = query.byContainer(CONTAINER_ID);
+        return client.execute(query).thenApply(response -> {
+                    Statistics statistics = new Statistics();
+                    if (response.getTotal() < 1) {
+                        LOG.error(String.format("No email to process for tenant %s", tenantConfiguration
+                                .getProjectKey()));
+                    }
+                    for (CustomObject<Email> customObject : response.getResults()) {
+                        Email email = customObject.getValue();
+                        String status = email != null?email.getStatus(): "";
+                        if (StringUtils.equalsIgnoreCase(status, EMAIL_STATUS_PENDING)) {
+                            int httpStatusCode = callWebHook(customObject.getId(), tenantConfiguration);
+                            statistics.update(httpStatusCode);
```

Comment:
Why do you need the response in the statistics? This statistics is just for you? Because the shops won't need this anyway. Maybe it could be faster if you just fire and forget the request?